addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6584,7 +6644,8 @@ registry.ImageTools = ImageHandlerOption.extend({
         await new Promise(resolve => {
             this.$target.one('image_cropper_destroyed', async () => {
                 if (isGif(this._getImageMimetype(img))) {
-                    img.dataset[img.dataset.shape ? 'originalMimetype' : 'mimetype'] = 'image/png';
+                    this.imageOptions[this.imageOptions.shape ? 'originalMimetype' : 'mimetype'] = 'image/png';
+                    weUtils.updateImageOptionsRegistry(img.getAttribute("src"), Object.assign({}, this.imageOptions));
```

Comment:
Maybe the proxy's `set` (and `delete`) could handle the call to update by itself ?
If you want the update to happen only once in a batch of changes, maybe a dedicated method on it with a callback could do the trick.
Something like:
```js
this.imageOptions.atomicUpdate(() => {
    delete this.imageOptions["xyz"];
    this.imageOptions["abc"] = 123;
});
```

Reply:
Unfortunately, I do not think it is possible: when updating the registry, we need to have the new source of the image (which is the original image on which the options have been applied) and the new set of options to be able to update the registry. Because the proxy is not aware of a change of src, I do not think it is possible that the proxy handles the call to update by itself.
In most cases, a change of options implies a change of the image src; that is the reason why the the update of the registry is done in the `_loadImage()` method. However, they are few cases (like here) that implies a change of options without a change of the image source. In those cases, the update of the registry has to be called explicitly.